### PR TITLE
Load metadata from our subgraph (metadataCID)

### PR DESCRIPTION
**Pull request by mtullyoc**
[Linear BAL-54](https://linear.app/bleu-llc/issue/BAL-54/load-metadata-from-our-subgraph-metadatacid)

Captura de Tela 2023-03-09 às 00 24 48

### CODE REVIEW

File: packages/next-app/src/lib/poolMetadataGql.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+const networkSdks = {
+  [Network.goerli]: goerliSdk,
+};
+
+const client = new GraphQLClient(ENDPOINTS[networkFor(currentNetwork)]);
+
+const gql = networkSdks[networkFor(currentNetwork)](client);
+
+export async function getPool(poolId: string) {
```
Comment (devjoaov) [inline packages/next-app/src/lib/poolMetadataGql.ts:31]: are we using this function?

**Review by devjoaov — APPROVED** [review]
LGTM. Thanks @mtullyoc 👍

File: packages/next-app/next.config.js [hunk 2]
```diff
@@ -1,7 +1,10 @@
 /** @type {import('next').NextConfig} */
 
 module.exports = {
-  transpilePackages: ["@balancer-pool-metadata/balancer-gql"],
+  transpilePackages: [
+    "@balancer-pool-metadata/balancer-gql",
+    "@balancer-pool-metadata/pool-metadata-gql",
```
Comment (ribeirojose) [inline packages/next-app/next.config.js:6]: `#comment` so much to write 🥵 starting to regret these names. But I don't have any better suggestions either 😅

File: packages/next-app/src/lib/gql.ts [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-  Network,
+  Network
 } from "@balancer-pool-metadata/balancer-gql/codegen";
-import { getSdkWithHooks as mainnetSdk } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/mainnet";
+import { getSdkWithHooks as arbitrumSdk } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/arbitrum";
 import { getSdkWithHooks as goerliSdk } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/goerli";
+import { getSdkWithHooks as mainnetSdk } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/mainnet";
 import { getSdkWithHooks as polygonSdk } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/polygon";
-import { getSdkWithHooks as arbitrumSdk } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/arbitrum";
+import { GraphQLClient } from "graphql-request";
```
Comment (ribeirojose) [inline packages/next-app/src/lib/gql.ts:9]: was this a lint rule? Can we have a task to automate that in CI checks?

File: packages/next-app/src/wagmi/generated.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// Generated by @wagmi/cli@0.1.11 on 3/7/2023 at 11:03:25 PM
+// Generated by @wagmi/cli@0.1.11 on 3/8/2023 at 11:53:55 PM
```
Comment (ribeirojose) [inline packages/next-app/src/wagmi/generated.ts:1]: `#comment` 🙄 starting to agree with @mendesfabio that this file probably shouldn't be committed. OR it shouldn't add this line here, AND it definitely should be in the linter/formatters ignore list.

File: packages/pool-metadata-gql/codegen.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  // arbitrum = "arbitrum",
+  goerli = "goerli",
+}
+
+export const ENDPOINTS = {
+  // [Network.mainnet]: `${BASE_URL}/balancer-v2`,
+  // [Network.polygon]: `${BASE_URL}/balancer-polygon-v2`,
+  // [Network.arbitrum]: `${BASE_URL}/balancer-arbitrum-v2`,
+  [Network.goerli]: `${BASE_URL}/balancer-pools-metadata`,
+};
```
Comment (ribeirojose) [inline packages/pool-metadata-gql/codegen.ts:17]: can you add a `#TODO(BAL-105): deploy to other networks` here?

**Review by ribeirojose — APPROVED** [review]

**Comment by ribeirojose** [comment]
Love the image 😆

**Review by mendesfabio — APPROVED** [review]
looking good

File: packages/pool-metadata-gql/codegen.ts (hunk 5 above)
Comment (mendesfabio) [reply packages/pool-metadata-gql/codegen.ts:17]: just to make sure everyone is on the same page - we'll only deploy to all networks once we finish the contracts (with support to DAO delegated pools) and ask for a review from Grants folks.

**Pull request merged**